pkg/metrics/metrics.go

```diff
@@ -33,10 +41,36 @@ var (
 		Name: "mapi_mao_collector_up",
 		Help: "Machine API Operator metrics are being collected and reported successfully",
 	}, []string{"kind"})
+
+	failedInstanceCreateCount = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "mapi_instance_create_failed",
+			Help: "Number of times provider instance create has failed.",
+		}, []string{"name", "namespace", "reason", "provider_name"},
+	)
+
+	failedInstanceUpdateCount = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "mapi_instance_update_failed",
+			Help: "Number of times provider instance update has failed.",
+		}, []string{"name", "namespace", "reason", "provider_name"},
+	)
+
+	failedInstanceDeleteCount = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "mapi_instance_delete_failed",
+			Help: "Number of times provider instance delete has failed.",
+		}, []string{"name", "namespace", "reason", "provider_name"},
+	)
 )
 
 func init() {
 	prometheus.MustRegister(MachineCollectorUp)
+	metrics.Registry.MustRegister(
+		failedInstanceCreateCount,
+		failedInstanceUpdateCount,
+		failedInstanceDeleteCount,
+	)
```

Comment:
We should be registering these all to the same registry, does the `MachineCollectorUp` metric show up in your tests?

Reply:
Opened https://issues.redhat.com/browse/OCPCLOUD-890 for that matter, and included https://issues.redhat.com/browse/OCPCLOUD-891 too.